lib/miq_pglogical.rb

```diff
@@ -0,0 +1,87 @@
+class MiqPglogical
+  REPLICATION_SET_NAME = 'miq'.freeze
+  SETTINGS_PATH = [:replication].freeze
+  NODE_PREFIX = "region_".freeze
+
+  def initialize(connection = ApplicationRecord.connection)
+    @connection = connection
+  end
+
+  # Configures the current database server to match the miq server configuration
+  def configure
+    mode = MiqServer.my_server.get_config.config.fetch_path(*SETTINGS_PATH, :mode)
+    case mode
+    when "local"
+      provider = MiqPglogicalProvider.new(@connection)
+      provider.configure_provider
+      provider.refresh_excludes
+    when "global"
+      subscriber = MiqPglogicalSubscriber.new(@connection)
+      subscriber.configure_subscriber
+      subscriber.refresh_subscriptions
+    when "none"
+      MiqPglogicalSubscriber.new(@connection).destroy_subscriber if subscriber?
+      MiqPglogicalProvider.new(@connection).destroy_provider if provider?
+      drop_node if node?
+    else
+      raise "'#{mode}' is not a valid replication mode"
+    end
+  end
+
+  # Returns whether or not this server is configured as a provider node
+  # @return Boolean
+  def provider?
+    pglogical.installed? && pglogical.enabled? && pglogical.replication_sets.include?(REPLICATION_SET_NAME)
```

Comment:
I feel like pglogical.enabled? should return false if it's not installed...then you don't have to manually specify both questions everywhere